### DISCUSSION TITLE
How to patch self-hosted postgresql database to sync up with Supabase

### DISCUSSION THREAD

**Discussion by kk21** [supabase/supabase#4394] [Questions]
Hi,

Does Supabase team provide SQL scripts to initialize and patch self-hosted postgresql database (various versions e.g. 13, 14) to fix Supabase bugs or work with the new versions of Supabase middleware.

Or is this in the roadmap.

Thanks.

**Comment by ChronSyn** [comment]
I believe the correct process would be to update the images, tear down the existing containers, and then launch new containers with the new images. If you're using a tool like Portainer, this is significantly easier than doing it via docker CLI.

In the `supabase/supabase` repo, there's a Docker folder (https://github.com/supabase/supabase/tree/master/docker) which includes various docker-compose files and dockerfiles used for initialising the Supabase stack, and SQL files used for initialising the database.

From what I've found, the JS lib has significant backwards compatibility with earlier versions of the server-side stack.

**Reply by kk21** [reply]
Thanks for helping.

Since the self-hosted postgresql database is decoupled from the Supabase middleware,
do you have any views if it is safe to re-run all the init scripts in the docker subfolder whenever necessary:
`https://github.com/supabase/supabase/tree/master/docker/volumes/db/init`
to fix or upgrade the Supabase "stuff" on the database (which could be any version such as 13 or 14)?

**Comment by philmas** [comment]
I am wondering whether version 15 for the postgres/supabase is actually supported. Does anyone know? I seem to get everything running except that I cannot connect from my frontend.
